plugins/gstreamer/src/Test.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Neurala Inc. 2013-2023
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:  The above copyright notice and this
+ * permission notice (including the next paragraph) shall be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <gst/gst.h>
+
+#include "GStreamerVideoSource.h"
+
+int main()
+{
+	char pipelineVariable[] = "NEURALA_GSTREAMER_PIPELINE=playbin uri=https://www.freedesktop.org/software/gstreamer-sdk/data/media/sintel_trailer-480p.webm video-sink=\"videoconvert ! video/x-raw,format=RGB,width=320,height=240 ! appsink name=mysink\"";
+	char sinkVariable[] = "NEURALA_GSTREAMER_SINK=mysink";
+	char widthVariable[] = "NEURALA_GSTREAMER_WIDTH=320";
+	char heightVariable[] = "NEURALA_GSTREAMER_HEIGHT=240";
```

Comment:
That should come from the DNN? Otherwise, prefer less env vars.

Reply:
In what way would that information be passed in? Plugin arguments?

Reply:
ping

Reply:
> In what way would that information be passed in?

Reply:
https://gstreamer.freedesktop.org/documentation/videoconvertscale/videoconvertscale.html

Reply:
Oh I remember why I did that. The point of the environment variables is not to provide that information to the pipeline (in fact, sticking it in the pipeline string is what I did in the test application with the  `video/x-raw,format=RGB,width=320,height=240` example caps). The environment variable thing is just a way for the plugin to know what dimensions to expect in `grabFrame`. I think it's possible to just get that from the caps of the `GstSample`, but I'd need to try that out.

Reply:
This might not work, though, because it's just reading the *capabilities* and not what the stream sample really is. Not sure in that case.